test/m1_facilitator/handlers/DeclaredDepositIntentsHandler/handle.test.ts

```diff
@@ -0,0 +1,198 @@
+import Message, { MessageType, MessageStatus } from '../../../../src/m1_facilitator/models/Message';
+import MessageRepository from '../../../../src/m1_facilitator/repositories/MessageRepository';
+import DepositIntentRepository from '../../../../src/m1_facilitator/repositories/DepositIntentRepository';
+import GatewayRepository from '../../../../src/m1_facilitator/repositories/GatewayRepository';
+import Repositories from '../../../../src/m1_facilitator/repositories/Repositories';
+
+describe('DeclaredDepositIntentsHandler::handle', (): void => {
+  let declaredDepositIntentsHandler: DeclaredDepositIntentsHandler;
+  let messageRepository: MessageRepository;
+  let depositIntentRepository: DepositIntentRepository;
+  let gatewayRepository: GatewayRepository;
+
+  const record = {
+    contractAddress: '0x0000000000000000000000000000000000000001',
+    messageHash: web3utils.sha3('2'),
+    intentHash: web3utils.sha3('10'),
+    tokenAddress: '0x0000000000000000000000000000000000000002',
+    beneficiary: '0x0000000000000000000000000000000000000050',
+    amount: new BigNumber(20),
+    feeGasLimit: new BigNumber(20),
+    feeGasPrice: new BigNumber(40),
+    blockNumber: new BigNumber('100'),
+  };
+  beforeEach(async (): Promise<void> => {
+    const repositories = await Repositories.create();
+    ({ depositIntentRepository, gatewayRepository, messageRepository } = repositories);
+    const gateway = new Gateway(
+      record.contractAddress,
+      '0x0000000000000000000000000000000000000010',
+      GatewayType.CONSENSUS,
+      '0x0000000000000000000000000000000000000001',
+      '0x0000000000000000000000000000000000000030',
+      new BigNumber(200),
+    );
+    await gatewayRepository.save(
+      gateway,
+    );
+
+    declaredDepositIntentsHandler = new DeclaredDepositIntentsHandler(
+      repositories.depositIntentRepository,
+      repositories.gatewayRepository,
+      repositories.messageRepository,
+    );
+  });
+
+  it('should change source status of message from undeclared to declared if message'
+    + ' is not present', async (): Promise<void> => {
+    await declaredDepositIntentsHandler.handle([record]);
+
+    const message = await messageRepository.get(record.messageHash);
+
+    assert.strictEqual(
+      message && message.type,
+      MessageType.Deposit,
+      'Message type should be deposit',
+    );
+
+    assert.strictEqual(
+      message && message.sourceStatus,
+      MessageStatus.Declared,
+      'Source status must be declared',
+    );
+
+    assert.strictEqual(
+      message && message.targetStatus,
+      MessageStatus.Undeclared,
+      'Target status must be undeclared',
+    );
+
+    const depositIntent = await depositIntentRepository.get(record.messageHash);
+
+    assert.isOk(
+      depositIntent,
+      `Record must be present for ${record.messageHash} in deposit intent model`,
+    );
+  });
+
+  it('should create a deposit intent if message is not present is'
```

Comment:
update the statement. there is repetition

Reply:
We don't need this it unit test. it's already asserted in above it.

Reply:
Removed it as there was assertion in the previous tests.